app/models/cart.rb

```diff
@@ -26,6 +34,14 @@ class Cart < ApplicationRecord
   has_one :coupon
   belongs_to :customer, optional: true
 
+  belongs_to :customer, optional: true
+  belongs_to :delivery, optional: true
+  belongs_to :credit_card, optional: true
+
+  after_create :set_number_and_status
+
+  enum status: %i[in_progress in_queue in_delivery delivered canceled]
```

Comment:
I'd suggest to explicitly map the relation between attribute and database integer with a hash:
```ruby
enum status: { in_progress: 0, in_queue: 1, in_delivery: 2, delivered: 3, canceled: 4]
```

This will strictly define which value in table will be for each status.

Reply:
I've taken into consideration your wise advice and defined enum statuses as a hash in order to explicitly bound statuses to their related values.